### PR TITLE
Make Analytics work with resources edges

### PR DESCRIPTION
**Pull request by duckofyork**

### CODE REVIEW

**Comment by duckofyork** [comment]
@aelred Can you please review this PR?

File: grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/DegreeStatisticsVertexProgram.java [hunk 1]
```diff
@@ -86,20 +87,26 @@ public boolean terminate(final Memory memory) {
     static void degreeStatisticsStepResourceOwner(Vertex vertex, Messenger<Long> messenger, Set<LabelId> ofLabelIds) {
         LabelId labelId = Utility.getVertexTypeId(vertex);
         if (labelId.isValid() && !ofLabelIds.contains(labelId)) {
-            messenger.sendMessage(messageScopeIn, 1L);
+            messenger.sendMessage(messageScopeShortcutIn, 1L);
+            messenger.sendMessage(messageScopeResourceOut, 1L);
```
Comment (aelred) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/DegreeStatisticsVertexProgram.java]: Don't we need to send both in _and_ out on resource edges?

File: grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/Utility.java [hunk 2]
```diff
@@ -109,4 +117,33 @@ static boolean isAlive(Vertex vertex) {
...
+    /**
+     * Get the resource edge id if there is one. Return null if not.
+     */
+    public static ConceptId getResourceEdgeId(GraknTx graph, ConceptId conceptId1, ConceptId conceptId2) {
+        if (mayHaveResourceEdge(graph, conceptId1, conceptId2)) {
+            List<Answer> answers = graph.graql().match(
+                    var("x").id(conceptId1),
+                    var("y").id(conceptId2),
+                    var("z").rel(var("x")).rel(var("y"))
+            ).select("z").execute();
```
Comment (aelred) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/Utility.java]: There's a method called `get` that might help here. You can do `query.get("z").forEach(concept -> ...)`

File: grakn-test/test-integration/src/test/java/ai/grakn/test/graql/analytics/AnalyticsTest.java [hunk 3]
```diff
@@ -69,28 +69,28 @@ public void setUp() {
         factory = context.factoryWithNewKeyspace();
     }
 
-    @Ignore // No longer applicable
+    @Ignore
```
Comment (aelred) [inline grakn-test/test-integration/src/test/java/ai/grakn/test/graql/analytics/AnalyticsTest.java]: Is this still meant to be ignored?

File: grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/DegreeStatisticsVertexProgram.java (hunk 1 above)
Comment (duckofyork) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/DegreeStatisticsVertexProgram.java]: No, the edge label is different

File: grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/Utility.java (hunk 2 above)
Comment (duckofyork) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/Utility.java]: cool. Let me change this.

File: grakn-test/test-integration/src/test/java/ai/grakn/test/graql/analytics/AnalyticsTest.java (hunk 3 above)
Comment (duckofyork) [inline grakn-test/test-integration/src/test/java/ai/grakn/test/graql/analytics/AnalyticsTest.java]: Yes, for now

**Comment by duckofyork** [comment]
Noooooooo!

**Comment by duckofyork** [comment]
@aelred finally...

**Pull request merged**